Feliz.CompilerPlugins/IntrinsicElementPrimitive.fs

```diff
@@ -0,0 +1,18 @@
+namespace Feliz
+
+open Fable
+open Fable.AST
+
+type IntrinsicElementPrimitiveAttribute(tagName:string) =
+    inherit MemberDeclarationPluginAttribute()
+    override _.FableMinimumVersion = "3.0"
+
+    override _.TransformCall(logger, memb, expr) =
+        match expr with
+        | Fable.Call(callee, info, typeInfo, range) when info.Args.Length = 1 ->
+            let element = AstUtils.makeStrConst tagName
+            AstUtils.makeCall (AstUtils.makeImport "createElement" "react") [element; AstUtils.nullValue; info.Args.[0]]
```

Comment:
You should be able to get the same effect just by inlining, this is what Fable.React does.

Reply:
Scratch that, we're going to need the plugin if we want to spread `children`.

Reply:
Yeah I am using this attribute specifically for when we have primitive values as the only children of the component:
```
Html.h1 "title" => <h1>title</h1> => createElement("h1", null, "title")
```
Spreading child components across `createElement` is a lot more complicated and it is what `[<IntrinsicElementChildren>]` is trying to accomplish at the moment. However, I am hitting some roadblocks when it comes to lists of components that have non-constant expressions like a conditional in the middle which changes the entire list expression into an ugly delayed sequence, wrapping the entire thing within a `List_fromSeq` function 😢

It is understandable from Fable perspective but it makes spreading the child components not always possible at compile-time so when the basic list pattern isn't detected to the apply the spread, the children are being constructed at runtime as a property

That is also why I can't inline, I want to apply the function body as is without transformation when I can't detect a certain pattern, this way the library stays working even when used from Fable 2.x

Reply:
If you cannot get the list items from the expression you can just spread it. There's no spread operator in Fable AST but you can emit JS (pseudocode): `Emit("...$0", [expr])`

Reply:
Wait, you can spread any arbitrary iterable like that?! 😱 I will need to play with this some more for sure